src/lib/statistics/chunk_statistics/histograms/abstract_histogram.hpp

```diff
@@ -0,0 +1,214 @@
+
+  /**
+   * Strings are internally transformed to a number, such that a bin can have a numerical width.
+   * This transformation is based on uint64_t.
+   */
+  using HistogramWidthType = std::conditional_t<std::is_same_v<T, std::string>, uint64_t, T>;
+
+  virtual HistogramType histogram_type() const = 0;
+
+  /**
+   * Returns a string with detailed information about the histogram, including the edges of the individual bins.
+   */
+  std::string description() const;
+
+  /**
+   * Returns the estimated selectivity given a predicate type and its parameter(s).
+   * It will always be between 0 and 1.
+   */
+  float estimate_selectivity(const PredicateCondition predicate_type, const AllTypeVariant& variant_value,
+                             const std::optional<AllTypeVariant>& variant_value2 = std::nullopt) const;
+
+  /**
+   * Returns the estimated cardinality given a predicate type and its parameter(s).
+   * It will always be between 0 and total_count().
+   * This method is specialized for strings to handle predicates uniquely applicable to string columns.
+   */
+  float estimate_cardinality(const PredicateCondition predicate_type, const AllTypeVariant& variant_value,
+                             const std::optional<AllTypeVariant>& variant_value2 = std::nullopt) const;
+
+  /**
+   * Returns whether a given predicate type and its parameter(s) can be pruned.
+   * This method is specialized for strings to handle predicates uniquely applicable to string columns.
+   */
+  bool can_prune(const PredicateCondition predicate_type, const AllTypeVariant& variant_value,
+                 const std::optional<AllTypeVariant>& variant_value2 = std::nullopt) const override;
+
+  /**
+   * Returns the lower bound (minimum value) of the histogram.
+   * This is equal to the smallest value in the segment.
+   */
+  T min() const;
+
+  /**
+   * Returns the upper bound (maximum value) of the histogram.
+   * This is equal to the largest value in the segment.
+   */
+  T max() const;
+
+  /**
+   * Returns the number of bins actually present in the histogram.
+   * This number can differ from the number of bins requested when creating a histogram.
+   */
+  virtual BinID bin_count() const = 0;
+
+  /**
+   * Returns the number of values represented in the histogram.
+   * This is equal to the length of the segment during creation, without null values.
+   */
+  virtual HistogramCountType total_count() const = 0;
+
+  /**
+   * Returns the number of distinct values represented in the histogram.
+   * This is equal to the number of distinct values in the segment during creation.
+   */
+  virtual HistogramCountType total_distinct_count() const = 0;
+
+ protected:
+  /**
+   * Returns a list of pairs of distinct values and their respective number of occurrences in a given segment.
+   * The list is sorted by distinct value from lowest to highest.
+   */
+  static std::vector<std::pair<T, HistogramCountType>> _value_counts_in_segment(
```

Comment:
Isn't this a histogram with a width of 1?
Also, maybe `_gather_value_distribution`?

Reply:
> Isn't this a histogram with a width of 1?

It is essentially the result of the query `SELECT c, COUNT(*) FROM segment GROUP BY c;`.
This is basically an `EqualElementCountHistogram` where each bin contains only one value.
Are you getting at something in particular?

> Also, maybe _gather_value_distribution?

Changed.

Reply:
>  Are you getting at something in particular?

Nope, just to make sure that I understood you right